api/resource/origin.go

```diff
@@ -8,31 +8,38 @@ import (
 	"strings"
 
 	"sigs.k8s.io/kustomize/api/internal/git"
+	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
 )
 
 // Origin retains information about where resources in the output
 // of `kustomize build` originated from
 type Origin struct {
 	// Path is the path to the resource, rooted from the directory upon
 	// which `kustomize build` was invoked
-	Path string
+	Path string `json:"path,omitempty" yaml:"path,omitempty"`
 
 	// Repo is the remote repository that the resource originated from if it is
 	// not from a local file
-	Repo string
+	Repo string `json:"repo,omitempty" yaml:"repo,omitempty"`
 
 	// Ref is the ref of the remote repository that the resource originated from
 	// if it is not from a local file
-	Ref string
+	Ref string `json:"ref,omitempty" yaml:"ref,omitempty"`
 }
 
 // Copy returns a copy of origin
 func (origin *Origin) Copy() Origin {
+	if origin == nil {
+		return Origin{}
+	}
 	return *origin
 }
 
 // Append returns a copy of origin with a path appended to it
```

Comment:
I'm a little bit unclear about this logic. If an origin is not initialized (nil), shall "Append" init the origin and then append the string, or raise nil errors rather than silently return Nil?

Reply:
>  If an origin is not initialized (nil), shall "Append" init the origin and then append the string

I changed the logic to do this!